shared/bls/bls.go

```
@@ -1,88 +1,107 @@
-// Package bls implements a go-wrapper around a C BLS library leveraging
-// the BLS12-381 curve. This package exposes a public API for verifying and
-// aggregating BLS signatures used by Ethereum Serenity.
+// Package bls implements a go-wrapper around a library implementing the
+// the BLS12-381 curve and signature scheme. This package exposes a public API for
+// verifying and aggregating BLS signatures used by Ethereum 2.0.
 package bls
 
 import (
 	"fmt"
-	"math/big"
+	"io"
+
+	gobls "github.com/phoreproject/bls"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
 )
 
 // Signature used in the BLS signature scheme.
-type Signature struct{}
+type Signature struct {
+	val *gobls.Signature
+}
 
-// SecretKey used in the BLS scheme.
+// SecretKey used in the BLS signature scheme.
 type SecretKey struct {
-	K *big.Int
+	val *gobls.SecretKey
 }
 
-// PublicKey corresponding to secret key used in the BLS scheme.
-type PublicKey struct{}
-
-// PublicKey returns the corresponding public key for the
-// Secret Key
-func (s *SecretKey) PublicKey() (*PublicKey, error) {
-	return &PublicKey{}, nil
+// PublicKey used in the BLS signature scheme.
+type PublicKey struct {
+	val *gobls.PublicKey
 }
 
-// BufferedSecretKey returns the secret key in a byte format.
-func (s *SecretKey) BufferedSecretKey() []byte {
-	return s.K.Bytes()
+// RandKey creates a new private key using a random method provided as an io.Reader.
+func RandKey(r io.Reader) (*SecretKey, error) {
+	k, err := gobls.RandKey(r)
+	if err != nil {
+		return nil, fmt.Errorf("could not initialize secret key: %v", err)
+	}
+	return &SecretKey{val: k}, nil
 }
 
-// BufferedPublicKey returns the public key in a byte format.
-func (p *PublicKey) BufferedPublicKey() []byte {
-	return []byte{}
+// SecretKeyFromBytes creates a BLS private key from a byte slice.
+func SecretKeyFromBytes(priv []byte) (*SecretKey, error) {
+	k := bytesutil.ToBytes32(priv)
```

could you return an error if `len(priv) != 32`,